### PR TITLE
[Enhancement] Add deactivate SSL certification verity option

### PR DESCRIPTION
**Pull request by kim3321**
Enables to disable ssl certificate checking

## Motivation

Add the option not to check ssl certificate to allow downloads to be performed in environments where ssl errors occur (for example, on-premises)

## Modification

Some codes have been modified so that the requests module can select whether to check the ssl certificate or not module.

## Use cases 

> mim download mmcls **--no-check-certificate** --config resnet18_8xb16_cifar10 --dest .

### CODE REVIEW

**Comment by ice-tong** [comment]
Hi @kim3321 , sorry for the late reply. ^_^

This changes looks grate to me, thanks for your contribution!

File: README.md [hunk 1]
```diff
@@ -174,6 +174,7 @@ Please refer to [installation.md](docs/en/installation.md) for installation.
   ```bash
   > mim download mmcls --config resnet18_8xb16_cifar10
   > mim download mmcls --config resnet18_8xb16_cifar10 --dest .
+  > mim download mmcls --no-check-certificate --config resnet18_8xb16_cifar10 --dest .
```
Comment (ice-tong) [inline README.md:177]: Since the `--no-check-certificate` is not a commonly used option. I don't think it's necessary to update the README. ^_^

Comment (kim3321) [reply README.md:177]: @ice-tong Thank you for your review!
Then, how to edit the code? I am contributed to github for the first time.

**Review by ice-tong — APPROVED** [review]
LGTM, thanks!

File: mim/commands/download.py [hunk 2]
```diff
@@ -47,17 +54,20 @@ def cli(package: str,
...
 def download(package: str,
              configs: List[str],
+             check_certificate: bool = True,
              dest_root: Optional[str] = None) -> List[str]:
     """Download checkpoints from url and parse configs from package.
 
     Args:
         package (str): Name of package.
         configs (List[str]): List of config ids.
+        check_certificate (bool): Whether to check the ssl certificate.
```
Comment (zhouzaida) [inline mim/commands/download.py:69]: Adding a new parameter in the middle position may cause a bc issue. Suggest moving it to the end.

File: mim/commands/download.py [hunk 3]
```diff
@@ -35,10 +35,17 @@
...
+    '--no-check-certificate',
+    'check_certificate',
+    is_flag=True,
+    default=True,
+    help='Ignore ssl certificate check')
 @click.option(
     '--dest', 'dest_root', type=str, help='Destination of saving checkpoints.')
 def cli(package: str,
         configs: List[str],
+        check_certificate: bool = True,
```
Comment (zhouzaida) [inline mim/commands/download.py:48]: Adding a new parameter in the middle position may cause a bc issue. Suggest moving it to the end.

File: mim/commands/download.py [hunk 4]
```diff
@@ -35,10 +35,17 @@
     cls=OptionEatAll,
     required=True,
     help='Config ids to download, such as resnet18_8xb16_cifar10')
+@click.option(
+    '--no-check-certificate',
```
Comment (zhouzaida) [inline mim/commands/download.py:39]: how about using `--verify/--no-verify'. The usage can be referred to https://github.com/open-mmlab/mim/blob/fae573e4aba32a96b95f94ee36561843e7a2314d/mim/commands/search.py#L58

Comment (kim3321) [reply mim/commands/download.py:39]: Then how about using '--ignore-ssl' instead?

File: mim/utils/utils.py [hunk 5]
```diff
@@ -157,19 +161,23 @@ def get_content_from_url(url: str,
 @typing.no_type_check
 def download_from_file(url: str,
                        dest_path: str,
+                       check_certificate: bool,
```
Comment (zhouzaida) [inline mim/utils/utils.py:164]: Adding a new parameter in the middle position may cause a bc issue. Suggest moving it to the end.

File: mim/commands/download.py (hunk 4 above)
Comment (zhouzaida) [reply mim/commands/download.py]: > Then how about using '--ignore-ssl' instead?

It sounds good.

File: mim/utils/utils.py [hunk 6]
```diff
@@ -157,19 +161,23 @@ def get_content_from_url(url: str,
 @typing.no_type_check
 def download_from_file(url: str,
                        dest_path: str,
-                       hash_prefix: Optional[str] = None) -> None:
+                       hash_prefix: Optional[str] = None,
+                       check_certificate: bool) -> None:
```
Comment (zhouzaida) [inline mim/utils/utils.py:165]:
```suggestion
                       check_certificate: bool = True) -> None:
```

File: mim/utils/utils.py (hunk 5 above)
Comment (kim3321) [reply mim/utils/utils.py]: I resolved it.

File: mim/utils/utils.py (hunk 6 above)
Comment (zhouzaida) [reply mim/utils/utils.py:165]: Please also update the order in the docstring

Comment (zhouzaida) [reply mim/utils/utils.py:165]: Thanks for your contribution. This PR can be merged after resolving the final comment.

Comment (kim3321) [reply mim/utils/utils.py]: Thank you for your review!

File: mim/commands/download.py [hunk 7]
```diff
@@ -112,7 +122,8 @@ def download(package: str,
                 echo_success(f'{filename} exists in {dest_root}')
             else:
                 # TODO: check checkpoint hash when all the models are ready.
-                download_from_file(checkpoint_url, checkpoint_path)
+                download_from_file(checkpoint_url, checkpoint_path,
+                                   check_certificate)
```
Comment (zhouzaida) [inline mim/commands/download.py:126]:
```suggestion
                                   check_certificate=check_certificate)
```

**Review by zhouzaida — APPROVED** [review]

**Comment by zhouzaida** [comment]
Hi, the lint failed.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**